core-rust/state-manager/src/state_manager.rs

```diff
@@ -805,12 +805,20 @@ where
             );
         }
 
-        let mut db_transaction = self.store.create_db_transaction();
+        // Create a transient substate store for all txns in the commit batch,
+        // backed by a StagedSubstateStore.
+        let mut commit_substate_store = CommitTransientSubstateStore::new(&mut self.store);
```

Comment:
With introduction of execution cache you will not be able to get a hold of `&mut self.store`. In that case you would have to construct `substates` from the retrieved `TransactionReceipt`s afterwards.

Reply:
Thanks for a heads-up! It shouldn't be a problem